circom-types/src/traits.rs

```diff
@@ -512,10 +522,30 @@ where
     const GT_SERIALIZED_BYTE_SIZE_COMPRESSED: usize;
     /// Size of uncompressed element of Gt in bytes
     const GT_SERIALIZED_BYTE_SIZE_UNCOMPRESSED: usize;
+    /// Deserializes element of G1 from bytes where the element is already in montgomery form (no montgomery reduction performed)
+    fn g1_from_bytes(bytes: &[u8]) -> IoResult<Self::G1Affine>;
```

Comment:
Why do we need those? &[u8] implements Reader so the other function should always work. Also this would then be a breaking change

Reply:
the reader one does one extra copy if I recall correctly, but this should hopefully be optimized away, so we can just use this one, you are right.